bin/concatenate_all.py

```diff
@@ -49,32 +49,31 @@ def parse_args():
     return parser.parse_args()
 
 
-def main():
-    start_time = time.time()
-    args = parse_args()
-
-    sys.stdout.write("CONCATENATE ALL...")  # don't want a newline here so can't use print
-    sys.stdout.flush()
-
-    config_path = os.path.join(args.basepath, "config.json")
-    if not os.path.isfile(config_path):
-        print(f"Error: invalid config path provided ({config_path})")
-        exit(1)
-
-    with open(config_path) as config_file:
-        config = json.load(config_file)
-
-    semester = config["semester"]
-    course = config["course"]
-    gradeable = config["gradeable"]
-    version_mode = config["version"]
-    users_to_ignore = config["ignore_submissions"]
+def validate(config, args):
+    # load parameters from the config to be checked
     regex_patterns = config["regex"].split(',')
     regex_dirs = config["regex_dirs"]
+    language = config["language"]
+    threshold = int(config["threshold"])
+    sequence_length = int(config["sequence_length"])
     prior_term_gradeables = config["prior_term_gradeables"]
 
-    # ==========================================================================
-    # Error checking
+    # Check we have a tokenizer to support the configured language
+    langs_data_json_path = "./data.json"  # data.json is in the Lichen/bin directory after install
+    with open(langs_data_json_path, 'r') as langs_data_file:
+        langs_data = json.load(langs_data_file)
+        if language not in langs_data:
+            print(f"\n\nERROR! tokenizing and hashing not supported for language {language}\n\n")
+            exit(1)
```

Comment:
So two things:
* Do not use a bare `exit()`. That's a python REPL construct, and not something that's built into the language itself. You're looking for [`sys.exit`](https://docs.python.org/3/library/sys.html#sys.exit) for quitting out of a program. 
* A more "pythonic" way to do this is to use [`SystemExit`](https://docs.python.org/3/library/exceptions.html#SystemExit), which is technically what `sys.exit` throws. It allows you write a bit condenser code that does the same thing:

```suggestion
            raise SystemExit(f"\n\nERROR! tokenizing and hashing not supported for language {language}\n\n")
```